src/services/standardGoals.ts

```diff
@@ -0,0 +1,243 @@
+  status: string[] = [
+    GOAL_STATUS.NOT_STARTED,
+    GOAL_STATUS.IN_PROGRESS,
+  ],
+) {
+  return Goal.findOne({
+    where: {
+      grantId,
+      goalTemplateId,
+      status: {
+        [Op.in]: status,
+      },
+    },
+    attributes: [
+      'id',
+      'name',
+      'status',
+      'goalTemplateId',
+      'grantId',
+    ],
+    include: [{
+      attributes: [
+        'id',
+        'title',
+        'onAR',
+        'status',
+        'objectiveTemplateId',
+      ],
+      model: Objective,
+      as: 'objectives',
+    }, {
+      model: GoalFieldResponse,
+      as: 'responses',
+      attributes: [
+        'id',
+        'goalId',
+        'response',
+      ],
+    }],
+  });
+}
+
+export async function getStardard(
+  standardGoalId: number,
+  grantId: number,
+  rootCauses?: Array<string>,
+) {
+  // Get the curated template with goal usage
+  const standard = await GoalTemplate.findByPk(standardGoalId, {
+    include: [
+      {
+        model: Goal,
+        as: 'goals',
+        where: {
+          grantId,
+          status: [
+            GOAL_STATUS.NOT_STARTED,
+            GOAL_STATUS.IN_PROGRESS,
+          ],
+        },
+        required: false,
+      },
+      {
+        model: GoalTemplateFieldPrompt,
+        as: 'prompts',
+        required: false,
+      },
+    ],
+  });
+
+  if (!standard) {
+    throw new Error('Standard goal not found');
+  }
+
+  const { prompts } = standard;
+  const requiresPrompts = prompts && prompts.length > 0;
+
+  // as above, if we ever need to add more prompt responses, we will need to make this generic
+  // (for the past few years, we've only had root causes,
+  // and have been told that is all we should expect)
+  if (requiresPrompts && !rootCauses) {
+    throw new Error('Root causes are required for this goal');
+  }
+
+  return { standard, requiresPrompts };
+}
+
+// This function will handle
+// - creating a new standard goal
+// - based on the design, this will not unsuspend or "restart" a closed goal
+// - creating new objectives if appropriate
+// - This will also serve to restart a standard goal (that has been closed)
+export async function newStandardGoal(
+  grantId: number,
+  standardGoalId: number,
+  objectives?: Array<IObjective>,
+  // todo: if we ever add more prompt responses, we will need to make this next param generic
+  rootCauses?: Array<string>,
+) {
+  const { standard, requiresPrompts } = await getStardard(standardGoalId, grantId, rootCauses);
+
+  if (standard.goals.length > 0) {
+    throw new Error('Standard goal has already been utilized');
+  }
+
+  const newGoal = await Goal.create({
+    status: GOAL_STATUS.NOT_STARTED,
+    name: standard.templateName,
+    grantId,
+    goalTemplateId: standard.id,
+    createdVia: 'rtr',
+  });
+
+  // a new goal does not require objectives, but may include them
+  if (objectives && objectives.length) {
+    await Objective.bulkCreate(objectives.map((objective) => ({
```

Comment:
Same here for objective template.

Reply:
here, we don't need the objectiveTemplate, this is only for creating a new goal w/ new objectives